app/javascript/packs/containers/NavigationBar/NavigationBarContainer.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+
+import {loadInitialState} from '../../actions/initialState';
+import {getBuildings, getBuildingById} from '../../selectors/buildingsSelector';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import DropdownMenuContainer from './DropdownMenuContainer'
+import {getName, getEmail} from '../../selectors/usersSelector'
+import {
+  getCurrentCategory,
+  getCategoriesForBuilding,
+  getFirstUnansweredCategory
+} from "../../selectors/categoriesSelector";
+import CategoryContainer from './CategoryContainer';
+import {getQuestionsByBuilding, getQuestionsByCategory} from "../../selectors/questionsSelector";
+import {getRemainingAnswersforCategory} from "../../selectors/answersSelector";
+
+
+class NavigationBarContainer extends React.Component {
+
+  render() {
+    const username = this.props.username;
+    const userEmail = this.props.userEmail;
+    return (
+      <div>
+        <DropdownMenuContainer
+          buildings={this.props.buildings}
+          currentBuilding={this.props.currentBuilding}
+          history={this.props.history}>
+        </DropdownMenuContainer>
+
+        <CategoryContainer
+          categories={this.props.categories}
+          currentCategory={this.props.currentCategory}
+          currentBuilding={this.props.currentBuilding}
+          remainingQuestions={this.props.remainingQuestions}>
+        </CategoryContainer>
+
+        <div
+          className="userInfo">
+          {username} <br/>
+          {userEmail}
+        </div>
+      </div>
+    )
+
+  }
+}
+
+
+function mapStateToProps(state, ownProps) {
+  //WE NEED THE ID TO BE A BUILDING ID SO WE CHECK ENTITY REPEATEDLY
```

Comment:
maybe put more comments about what the below is doing? I feel like a lot of the logic for questions remaining should maybe be put into a separate util file. Selectors are supposed to pick things off the redux tree in a relatively simple way, and something as complex as this should maybe be broken out into an importable function where you pass in whole object collections as parameters. So maybe like a `getRemainingQuestions()` functions that takes in `state.questions` by building or something that's easier to select.

Reply:
@arpanpal Is it possible that we can just default `remainingQuestions` to 0 in the beginning, and have `CategoryContainer` figure out if we are in a building view? (if not, don't show `remainingQuestions` anyway) 
As Helen mentioned, inline docs would be a nice-to-have as well.

@HacksForHugs would that make things more straightforward?

Reply:
then i would have to pass down the properties of match, is that what we wanna do? I'll def do some inline documentation